programs/obfuscator/Obfuscator.cpp

```diff
@@ -1287,32 +1291,57 @@ try
     markov_model_params.determinator_sliding_window_size = options["determinator-sliding-window-size"].as<UInt64>();
 
     /// Create the header block
-    std::vector<std::string> structure_vals;
-    boost::split(structure_vals, structure, boost::algorithm::is_any_of(" ,"), boost::algorithm::token_compress_on);
-
-    if (structure_vals.size() % 2 != 0)
-        throw Exception("Odd number of elements in section structure: must be a list of name type pairs", ErrorCodes::LOGICAL_ERROR);
+    SharedContextHolder shared_context = Context::createShared();
+    auto context = Context::createGlobal(shared_context.get());
+    context->makeGlobalContext();
 
     Block header;
-    const DataTypeFactory & data_type_factory = DataTypeFactory::instance();
 
-    for (size_t i = 0, size = structure_vals.size(); i < size; i += 2)
+    if (structure.empty())
     {
-        ColumnWithTypeAndName column;
-        column.name = structure_vals[i];
-        column.type = data_type_factory.get(structure_vals[i + 1]);
-        column.column = column.type->createColumn();
-        header.insert(std::move(column));
+        ReadBufferIterator read_buffer_iterator = [&]()
+        {
+            return std::make_unique<ReadBufferFromFileDescriptor>(STDIN_FILENO);
```

Comment:
I see you removed using `PeekableReadBuffer`, why? We should come back to the beginning of the data after the schema is inferred

Reply:
Oh, I see, we expect that stdin is seekable

Reply:
Yep, we check it a bit later. But we can also add the same check with `lseek` here.